app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,61 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts
+    @posts = Post.all
   end
 
   def show
     @user = User.find(params[:user_id])
-    @post = @user.posts.find(params[:id])
+    @post = Post.find(params[:id])
+    @comment = Comment.new
+  end
+
+  def new
+    @current_user = User.first
+    @post = Post.new
+  end
+
+  def create
+    @current_user = User.first
+    @post = Post.new(
+      title: params[:post][:text],
+      text: params[:post][:text],
+      author: @current_user
+    )
+
+    if @post.save
+      redirect_to user_post_path(@current_user, @post)
+    else
+      flash[:error] = 'Post could not be saved'
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def create_comment
+    @current_user = User.first
+
+    post = Post.find(params[:id])
+    comment = Comment.new(
+      text: params[:comment][:text],
+      author: @current_user,
+      post:
+    )
+
+    if comment.save
+      redirect_to user_post_path(post.author, post)
+    else
+      flash[:error] = 'Comment could not be saved'
+      render :new, status: :unprocessable_entity
+    end
+  end
```

Comment:
- [ ] Instead of having the `#create_comment` method here in the `posts_controller`, please create a comments_controller which should have new and create actions.
- [ ] Kindly ensure to define a pair of routes in the config/routes.rb file for creating comments (the new and create actions)